do-not-commit/default-mutable-list.py

```diff
@@ -0,0 +1,392 @@
+import copy
+
+# ruleid: default-mutable-list
+def append_func1(default=[]):
+    default.append(5)
+
+
+# ruleid: default-mutable-list
+def append_func2(default=[]):
+    for x in range(10):
+        default.append(x)
+
+
+# todoruleid: default-mutable-list
+def append_func3(default=[]):
+    x = default
+    x.append(5)
+
+
+# ruleid: default-mutable-list
+def append_func4(x=1, default=[]):
+    default.append(5)
+
+
+# ruleid: default-mutable-list
+def append_func5(default=[]):
+    if not default:
+        default.append(1)
+
+
+# ruleid: default-mutable-list
+def append_func6(default=[], x="string"):
+    default.append(5)
+
+
+# todoruleid: default-mutable-list
+def append_func7(default=[]):
+    if True:
+        default = list(default)
+    else:
+        default.append(1)
+
+
+# ruleid: default-mutable-list
+def append_func8(default=[]):
```

Comment:
Function append_func8 mutates default list default. Instances of default function arguments in Python are shared between function calls.  Mutating a default function argument changes the result for all function consumers. This will most likely cause unexpected results, or lead to security vulnerabilities whereby one function consumer can either view or modify the data of another function consumer. ([.common-mistakes.default-mutable-list.default-mutable-list](https://semgrep.dev/r/.common-mistakes.default-mutable-list.default-mutable-list))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.common-mistakes.default-mutable-list.default-mutable-list) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>